DirectProgramming/DPC++FPGA/Tutorials/Features/device_global/README.md

```diff
@@ -0,0 +1,225 @@
+
+`device_globals` can be used to store state across multiple re-launches of a kernel without having to pass in a `buffer` as a kernel argument. This is useful in cases where components are nodes in a state-machine.
+
+### Non-zero-initialization of a `device_global` - pre C++20
+Without C++20 `consteval` keyword support, `device_globals` are zero-initialized. If there is a known first usage of a `device_global` in device code, there exists a technique, described below, that allows a `device_global` to be initialized to a non-zero value. 
+
+Using a second `device_global<bool>` representing a flag that will control when to initialize the `device_global` to a non-zero value. This flag will get zero-initialized to `false`. Once initialization happens, the flag is set to `true` and initialization code doesn't execute on subsequent relaunches of the kernel.
+
+```cpp
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+
+device_global<int> val;
+device_global<bool> is_val_initialized;
+
+int main () {
+  queue Q;
+  Q.submit([&](sycl::handler& h) {
+    h.single_task([=] {
+      // Initialization happens only once
+      if (!is_val_initialized) {
+        // do initialization
+        is_val_initialized.get() = true;
+      }
+
+      // uses of `val`
+    });
+  });
+}
+```
+
+
+### Non-zero-initialization of a `device_global`- with C++20
+In a future version of this extension, it is expected that when C++20 support is available and enabled, the `consteval` keyword will be used to enable compile-time constant initialization of the device allocations backing device_global. This will simplify some coding patterns, compared with the current zero-initialization requirement.
+
+```cpp
+using namespace sycl::ext::oneapi::experimental;
+
+device_global<int> val {42};
+```
+
+### Additional Documentation
+- [SYCL `device_global` Language Specification](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc) helps understand the API and of `device_globals` and restrictions on it's usage
+- [Explore SYCL* Through Intel&reg; FPGA Code Samples](https://software.intel.com/content/www/us/en/develop/articles/explore-dpcpp-through-intel-fpga-code-samples.html) helps you to navigate the samples and build your knowledge of FPGAs and SYCL.
+- [FPGA Optimization Guide for Intel&reg; oneAPI Toolkits](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide) helps you understand how to target FPGAs using SYCL and Intel&reg; oneAPI Toolkits.
+- [Intel&reg; oneAPI Programming Guide](https://software.intel.com/en-us/oneapi-programming-guide) helps you understand target-independent, SYCL-compliant programming using Intel&reg; oneAPI Toolkits.
+
+## Key Concepts
+* The basic usage of the `device_global` class
+* How initialize a `device_global` to non-zero value
+
+
+## Using Visual Studio Code*  (Optional)
```

Comment:
Huh? I copied this file from another code sample - so that code sample has the exact same problem. I think the tutorial is `private_copies`

Reply:
Do you have an example code sample that has them in teh right order?

Reply:
Hum interesting, it seems that the order is not the same in all samples :/ 
Don't worry about it, I'm trying the convince the code sample committee to move these repeated sections to a top level readme, so this would end up being remove hopefully :)